aiida/common/datastructures.py

```diff
@@ -104,6 +104,14 @@ class CodeInfo(DefaultFieldsAttributeDict):
 
         code.x cmdline_params[0] cmdline_params[1] ... < stdin > stdout
 
+    * ``folder_path``: (optional) the path, in string, of the folder where to
+      run the code. If it is specified, the code will be called as:
+
+        (cd folder_path; code.x ...)
+
+      The outputs will be produced in this folder.
+      NOTE: It is assumed that the folder already exists.
```

Comment:
Is there a reason why the engine cannot create it along with all the other input files?

Reply:
Maybe the way I wrote it is confusing. I meant that if one specify this option, in the CalcJob one should take care of creating the folder and writing the input file(s) in it. Maybe it was an unnecessary note, but just wanted to point out that if one specify this option one has to remember to create the (sub)folder and put the inputs in it while developing.

Reply:
I see, in the typical use case, the plugin would also write the input files in the relative directory so would have to create it anyway. Maybe a slightly rephrase would help.
```
The engine expects that the `CalcJob` plugin creates the relative path in which the code should be executed.
```
Or something to that effect.